api/helpers.go

```diff
@@ -9,8 +9,13 @@ import (
 
 func sendJSON(w http.ResponseWriter, status int, obj interface{}) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(status)
-	if err := json.NewEncoder(w).Encode(obj); err != nil {
+	b, err := json.Marshal(obj)
+	if err != nil {
 		logrus.WithError(err).Errorf("Error encoding json response: %v", obj)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(`{"code":500,"msg":"Error encoding json response: ` + err.Error() + `"}`))
```

Comment:
Which in turn calls `sendJSON`. If the error is because of some internal state issue, that seems like it would turn into an infinite loop.